### PR TITLE
DWR-1031

### PR DESCRIPTION
**Pull request by malaffoon**
Turns out none of this is really necessary since the problem was a configuration issue but the logging and the actuator end-point will be useful so i'm asking to commit anyway ...

### CODE REVIEW

File: report-processor/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# docker-compose specs for RDW report processor in a container and MySQL running natively (outside the containers).
```
Comment (malaffoon) [inline report-processor/docker-compose.yml]: Not sure we want this but i wanted to run just the report-processor (with config and rabbit) and the only docker-compose file was in the reporting webapp folder.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/ReportProcessorApplication.java [hunk 2]
```diff
@@ -5,16 +5,13 @@
...
-import org.springframework.context.annotation.Import;
 import org.springframework.context.annotation.PropertySource;
-import org.springframework.scheduling.annotation.EnableScheduling;
 
 @SpringBootApplication(exclude = {
         ElastiCacheAutoConfiguration.class,
         ContextStackAutoConfiguration.class
 })
 @EnableCaching
-@EnableScheduling
```
Comment (malaffoon) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/ReportProcessorApplication.java]: I did this when grasping for straws but it seems like it is okay to move it to the task configuration.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java [hunk 3]
```diff
@@ -1,17 +1,60 @@
...
     @Bean
-    public ThreadPoolTaskScheduler threadPoolTaskScheduler() {
-        final ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
-        threadPoolTaskScheduler.setPoolSize(5);
-        threadPoolTaskScheduler.setThreadNamePrefix("RemoveStaleReportsThread");
-        return threadPoolTaskScheduler;
+    public RemoveStaleReportsTask removeStaleReportsTask(
+            final ReportRemovalService reportRemovalService,
+            final TaskScheduler taskScheduler,
+            @Value("${task.remove-stale-reports.cron}") final String cron,
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java]: No `@ConfigurationProperties(prefix = "task.remove-stale-reports") RemoveStaleReportsSettings`?

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java [hunk 4]
```diff
@@ -1,17 +1,60 @@
...
+import org.springframework.cloud.endpoint.GenericPostableMvcEndpoint;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.EnableScheduling;
 
 @Configuration
+@EnableScheduling
+@ConditionalOnProperty(prefix = "task.remove-stale-reports", name = {"cron", "max-report-lifetime-days", "max-random-minutes"})
```
Comment (malaffoon) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java]: Changed this to be more like the task configurations in ingest where there is a main configuration class that controls all bean creation for the task. This eliminates instantiating components that aren't going to be used (if the task isn't enabled). Yeah, task is probably always going to be enabled but ...

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/web/RemoveStaleReportsEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.endpoint.AbstractEndpoint;
+
+/**
+ * This actuator end-point allows the stale report task to be manually triggered.<br/>
+ *
+ * It does not have a stereotype annotation because it is instantiated conditionally in
+ * {@link org.opentestsystem.rdw.reporting.processor.configuration.RemoveStaleReportsConfiguration}.<br/>
+ */
+public class RemoveStaleReportsEndpoint extends AbstractEndpoint<Boolean> {
```
Comment (malaffoon) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/web/RemoveStaleReportsEndpoint.java]: new thing

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalServiceTest.java [hunk 6]
```diff
@@ -29,18 +29,18 @@
 
     @Test
     public void itShouldRemoveOldReports() {
-        List<Report> foundReports = Arrays.asList(report);
+        List<Report> foundReports = Collections.singletonList(report);
```
Comment (malaffoon) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalServiceTest.java]: unrelated warning fix ... IntelliJ insisted this was mo' better.

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTaskTest.java [hunk 7]
```diff
@@ -1,55 +1,38 @@
...
-    @Mock
     private ReportRemovalService reportRemovalService;
-    @Mock
-    private Report report;
-    private ThreadPoolTaskScheduler taskScheduler;
+    private TaskScheduler taskScheduler;
 
     @Before
-    public void setup() {
-        taskScheduler = new ThreadPoolTaskScheduler();
```
Comment (malaffoon) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTaskTest.java]: simplified this unit test: there is no need to test the scheduler implementation, we just want to know that it is invoked with the proper arguments

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java (hunk 3 above)
Comment (malaffoon) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java]: I thought about it but it was just when i noticed the real problem was just configuration so i wanted to wrap this up.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java [hunk 8]
```diff
@@ -1,17 +1,60 @@
...
+     * It supports POST instead of GET, e.g.:<pre>
+     * curl -X POST http://localhost:8008/removeStaleReports
+     * </pre>
+     *
+     * @param reportRemovalService  service
+     * @param maxReportLifetimeDays config
+     * @return task end-point
+     */
+    @Bean
+    public MvcEndpoint reconciliationReportEndpoint(final ReportRemovalService reportRemovalService,
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java]: This is interesting. Why not a Controller?

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalService.java [hunk 9]
```diff
@@ -1,28 +1,34 @@
...
+ * {@link org.opentestsystem.rdw.reporting.processor.configuration.RemoveStaleReportsConfiguration}.<br/>
+ */
+public class DefaultReportRemovalService implements ReportRemovalService {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultReportRemovalService.class);
 
     private final ReportService reportService;
     private final ReportContentService reportContentService;
 
-    @Autowired
-    public DefaultReportRemovalRemovalService(final ReportService reportService, final ReportContentService reportContentService) {
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalService.java]: I liked the old name better : j

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTask.java [hunk 10]
```diff
@@ -3,46 +3,47 @@
...
+                                  final TaskScheduler taskScheduler,
+                                  final int maxReportLifetimeDays,
+                                  final int maxRandomMinutes) {
         this.reportRemovalService = reportRemovalService;
         this.taskScheduler = taskScheduler;
+        this.maxReportLifetimeDays = maxReportLifetimeDays;
+        this.maxRandomMinutes = maxRandomMinutes;
     }
 
     @Scheduled(cron = "${task.remove-stale-reports.cron}", zone = "GMT")
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTask.java]: just wondering - would anyone ever want to configure the time zone?

File: report-processor/docker-compose.yml (hunk 1 above)
Comment (esotrope) [reply report-processor/docker-compose.yml]: 👍

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java (hunk 8 above)
Comment (malaffoon) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/configuration/RemoveStaleReportsConfiguration.java]: It is more of an ops hook than an end-user thing so i want it to be an actuator end-point. And this is how they work.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalService.java (hunk 9 above)
Comment (malaffoon) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportRemovalService.java]: lol

I stared at the name for about 30 seconds trying to figure out if i was missing something.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTask.java (hunk 10 above)
Comment (malaffoon) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/tasking/RemoveStaleReportsTask.java]: Perhaps but ... tough. We're taking the stance that servers operate in GMT.
